src/Diagnostics/FxCop/Desktop.Analyzers/Core/Security/DoNotCatchCorruptedStateExceptionsAnalyzer.cs

```diff
@@ -0,0 +1,157 @@
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Security,
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: true,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: null,
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        protected abstract Analyzer GetAnalyzer(CompilationSecurityTypes compilationTypes, ISymbol owningSymbol, SyntaxNode codeBlock);
+
+        private static readonly ImmutableArray<DiagnosticDescriptor> s_supportedDiagnostics = ImmutableArray.Create(Rule);
+
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        {
+            get
+            {
+                return s_supportedDiagnostics;
+            }
+        }
+
+        /// <summary>
+        /// Initialize the analyzer.
+        /// </summary>
+        /// <param name="analysisContext">Analyzer Context.</param>
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.RegisterCompilationStartAction(
+                (context) =>
+                {
+                    var compilationTypes = new CompilationSecurityTypes(context.Compilation);
+                    if (compilationTypes.HandleProcessCorruptedStateExceptionsAttribute != null)
+                    {
+                        context.RegisterCodeBlockStartAction<TLanguageKindEnum>(
+                        codeBlockStartContext =>
+                        {
+                            ISymbol owningSymbol = codeBlockStartContext.OwningSymbol;
+                            if (owningSymbol.Kind == SymbolKind.Method)
+                            {
+                                var method = (IMethodSymbol)owningSymbol;
+
+                                ImmutableArray<AttributeData> attributes = method.GetAttributes();
+                                if (attributes.FirstOrDefault(attribute => attribute.AttributeClass == compilationTypes.HandleProcessCorruptedStateExceptionsAttribute) != null)
+                                {
+                                    Analyzer analyzer = GetAnalyzer(compilationTypes, owningSymbol, codeBlockStartContext.CodeBlock);
```

Comment:
Thank you for changing the analyzer implementation, this looks a lot more performant way to do it!